cmd/create.go

```diff
@@ -137,104 +197,333 @@ type createConfig struct {
+	templates, err := client.Templates().List(runtime)
+	if err != nil {
+		return err
+	}
+	for _, v := range templates {
+		fmt.Fprintf(&b, "  %v\n", v)
+	}
+	return errors.New(b.String())
+}
+
+// Validate the current state of the config, returning any errors.
+// Note this is a deeper validation using a client already configured with a
+// preliminary config object from flags/config, such that the client instance
+// can be used to determine possible values for runtime, templates, etc.  a
+// pre-client validation should not be required, as the Client does its own
+// validation.
+func (c createConfig) Validate(client *fn.Client) (err error) {
+	// Client validates both language runtime and template exist, defaulting
+	// them if not (to 'node' and 'http', respectively).  However, if either of
+	// them are invalid, or the chosen combination does not exist, the error
+	// message is a rather terse one-liner.  This is suitable for libraries, but
+	// for a CLI it behooves us to be more verbose, including valid options for
+	// each.  So here, we check that the values entered (if any) are both valid
+	// and valid together.
+	if c.Runtime != "" && !isValidRuntime(client, c.Runtime) {
+		return newInvalidRuntimeError(client, c.Runtime)
+	}
+
+	if c.Template != "" && !isValidTemplate(client, c.Runtime, c.Template) {
+		return newInvalidTemplateError(client, c.Runtime, c.Template)
+	}
+
+	// TODO: refactor to be git-like with no name at time of creation, but rather
+	// with named deployment targets in a one-to-many configuration.
+	dirName, _ := deriveNameAndAbsolutePathFromPath(c.Path)
+	return utils.ValidateFunctionName(dirName)
+	// NOTE: perhaps additional validation would be of use here in the CLI, but
+	// the client libray itself is ultimately responsible for validating all input
+	// prior to exeuting any requests.
+}
+
+// prompt the user with value of config members, allowing for interactively
+// mutating the values. The provided clientFn is used to construct a transient
+// client for use during prompt autocompletion/suggestions (such as suggesting
+// valid templates)
+func (c createConfig) prompt(client *fn.Client) (createConfig, error) {
+	var qs []*survey.Question
+
+	runtimes, err := client.Runtimes()
+	if err != nil {
+		return createConfig{}, err
+	}
+
+	// First ask for path...
+	qs = []*survey.Question{
 		{
 			Name: "path",
 			Prompt: &survey.Input{
-				Message: "Project path:",
+				Message: "Function Path:",
 				Default: c.Path,
 			},
 			Validate: func(val interface{}) error {
 				derivedName, _ := deriveNameAndAbsolutePathFromPath(val.(string))
 				return utils.ValidateFunctionName(derivedName)
 			},
-		},
-		{
-			Name: "runtime",
+			Transform: func(ans interface{}) interface{} {
+				_, absolutePath := deriveNameAndAbsolutePathFromPath(ans.(string))
+				return absolutePath
+			},
+		}, {
+			Name: "language", // users refer to this as Language Runtime (language for short)
```

Comment:
I believe this is actually not user facing, and instead should remain `"runtime"`. The `SurveyQuestion.Name` field is used to set the `createConfig` struct's value. In this case, it's `createConfig.Runtime`. 

https://github.com/knative-sandbox/kn-plugin-func/pull/547/files#diff-e52f1f4fa0c533e3287cbe2ef0436582b5ca471a3fddcdf1f49fecd1ec0c7d13R180

When I run the binary built from this branch, I get this error.

```
❯ kn func create -c
? Function Path: foobar
? Language Runtime: nodejs
Error: could not find field matching language
Error: exit status 1
```

By changing `"language"` to `"runtime"` here, I can successfully create a project.

Reply:
That was an incorrect 11th hour update.  Thank you.